R/MsBackendDataFrame-functions.R

```diff
@@ -238,3 +233,75 @@ MsBackendDataFrame <- function() {
             do.call(.SPECTRA_DATA_COLUMNS[column], args = list())
     } else stop("column '", column, "' not available")
 }
+
+#' @description
+#'
+#' Helper to be used in the filter functions to select the file/origin in
+#' which the filtering should be performed.
+#'
+#' @param object `MsBackend`
+#'
+#' @param dataStorage `character` or `integer` with either the names of the
+#'     `dataStorage` or their index (in `unique(object$dataStorage)`) in which
+#'     the filtering should be performed.
+#'
+#' @param dataOrigin same as `dataStorage`, but for the `dataOrigin` spectra
+#'     variable.
+#'
+#' @return `logical` of length equal to the number of spectra in `object`.
+#'
+#' @noRd
+.sel_file <- function(object, dataStorage = integer(), dataOrigin = integer()) {
```

Comment:
Do you still want to support indices here?

Reply:
I just wanted to keep the function for now - shouldn't hurt